org.scala-ide.sdt.core/src/org/scalaide/core/internal/extensions/autoedits/ApplyTemplateCreator.scala

```diff
@@ -0,0 +1,13 @@
+package org.scalaide.core.internal.extensions.autoedits
+
+import org.scalaide.core.text.Document
+import org.scalaide.core.text.TextChange
+import org.scalaide.extensions.autoedits.ApplyTemplate
+
+object ApplyTemplateCreator {
+  def create(doc: Document, change: TextChange): ApplyTemplate =
+    new ApplyTemplate {
```

Comment:
Yeah, they don't lead to the best code. But so far I don't know how to replace them. case classes do not work, because I want to be able to use path dependent types, which are required for the compiler. Compiler support is not required for auto edits, but I want to keep the design similar to save actions. I think instead of the create methods, a dependency injection approach similar to the ones that are used in the Java world could work better (mutable instance variables which are injected by the IDE). My original plan was to create auto edits & save actions dynamically, i.e. create instances by a toolbox & reflection. But I didn't find the time so far to complete this work.